docs/README.md

```diff
@@ -1,35 +0,0 @@
-# Documentation
```

Comment:
Why is this being deleted?

Reply:
Hey @bzbarsky-apple , several reasons:

- The page is mostly duplicating the page hierarchy ToC once Matter docs are deployed to github.io (see https://gautesl.github.io/connectedhomeip/docs/index.html )
- The page doesn't fit into the hierarchy as it is a duplicate of an index page, and requires manual maintenance.
- Keeping the page without including it in the doc framework (if possible at all) would also mean there is a parallel documentation to the one on github.io, precisely something we wanted to fix with this PR.

Reply:
And separately, having the documentation not be discoverable/browsable from the repo itself without a separate tool has some drawbacks.  What are the benefits we gain in return?  The PR description has a lot of "what" in terms of the changes but nothing regarding the "why".  Please add the reasons we want this change to the PR description; that will aid greatly in review.

Reply:
After some discussion with @greg-fer and @LuDuda we found it best to keep `docs/README.md` as it is, but not include it in the Sphinx-generated output. The top-level `README.md` should become the new landing page for the Sphinx-generated output, and the links there will be fixed.

I have updated the PR description with the main goals that this PR tries to achieve, but another benefit of using Sphinx is that it discovers broken links when building. As a result, this PR includes fixes for a bunch of existing broken links. The reason it did not pick up the broken links in the top-level `README.md` is that it was not included as a source to Sphinx.

Reply:
@gautesl Thank you; that makes sense.

I don't have strong views on `docs/README.md` per se.  I do want us to make sure the toplevel README does not have broken links.